juju/juju
# The storage Constraints struct is renamed to Directive.

Some time ago, the terminology "storage constraints" was used internally when we should have used "storage directive". This PR does that rename, primarily renaming the Constraints struct in `internal/storage` to "Directive".

This is mostly a mechanical change. Legacy state is left untouched and still uses constraints terminology. Legacy state will be going away soon.

The only non mechanical bit is model imports for migration. We use an updated `juju/description` package to supply the storage directives from an imported model, for both legacy models and newer ones.

## QA steps

bootstrap 
deploy postgresql with storage for pgdata
refresh the charm with different storage directive
add a unit, ensure new storage is used

repeat above with an older juju cli

I wanted to migrate a 3.x model to check that storage directives come across but we don't support migrating from 3.x to 4.x yet.

## Links

**Jira card:** JUJU-5706
